src/main/java/com/wavefront/sdk/entities/histograms/WavefrontHistogramImpl.java

```diff
@@ -266,19 +178,38 @@ private long currentMinuteMillis() {
   }
 
   /**
-   * Helper to retrieve the current bin. Will be invoked on the thread local histogramBinsList.
+   * Helper to get the current bin.
+   * Will flush currentMinuteBin into globalHistogramBinsList if it's a new minute.
    */
-  private MinuteBin getCurrentBin() {
-    ConcurrentLinkedDeque<MinuteBin> sharedBinsInstance = histogramBinsList.get();
+  private ThreadMinuteBin getCurrentBin() {
     long currMinuteMillis = currentMinuteMillis();
-    if (sharedBinsInstance.isEmpty() ||
-        sharedBinsInstance.getLast().minuteMillis != currMinuteMillis) {
-      sharedBinsInstance.add(new MinuteBin(ACCURACY, currMinuteMillis));
-      if (sharedBinsInstance.size() > MAX_BINS) {
-        sharedBinsInstance.removeFirst();
+    return flushCurrentBin(currMinuteMillis);
+  }
+
+  private ThreadMinuteBin flushCurrentBin(long currMinuteMillis) {
+    if (this.currentMinuteBin.minuteMillis == currMinuteMillis) return this.currentMinuteBin;
+    synchronized(this) {
+      if (this.currentMinuteBin.minuteMillis != currMinuteMillis) {
```

Comment:
Add a comment as to why double check inside synchronized (this) block is important to do here.
if (this.currentMinuteBin.minuteMillis != currMinuteMillis) { ...